source/material_model/utilities.cc

```diff
@@ -538,29 +552,76 @@ namespace aspect
-              std::string label;
-              in >> label;
-              if (label == "rho,kg/m3")
+              first_natural_variable = column_name;
+              in >> column_name;
+              AssertThrow(column_name == "T(K)", ExcMessage("The second column name in PerpleX lookup file " + filename + " should be T(K)."))
+            }
+          else if (column_name == "T(K)")
+            {
+              first_natural_variable = column_name;
+              in >> column_name;
+              AssertThrow(column_name == "P(bar)", ExcMessage("The second column name in PerpleX lookup file " + filename + " should be T(K)."))
+            }
+          else
+            {
+              AssertThrow(false, ExcMessage("The first column name in PerpleX lookup file " + filename + " should be P(bar) or T(K)."))
+            }
+
+          for (unsigned int n=2; n<n_columns; n++)
+            {
+              in >> column_name;
+              if (column_name == "rho,kg/m3")
                 prp_indices[0] = n;
-              else if (label == "alpha,1/K")
+              else if (column_name == "alpha,1/K")
                 prp_indices[1] = n;
-              else if (label == "cp,J/K/kg")
+              else if (column_name == "cp,J/K/kg")
                 prp_indices[2] = n;
-              else if (label == "vp,km/s")
+              else if (column_name == "vp,km/s")
                 prp_indices[3] = n;
-              else if (label == "vs,km/s")
+              else if (column_name == "vs,km/s")
                 prp_indices[4] = n;
-              else if (label == "h,J/kg")
+              else if (column_name == "h,J/kg")
                 prp_indices[5] = n;
+              else if (column_name.length() > 3)
+                {
+                  if (column_name.substr(0,13).compare("vol_fraction_") == 0)
+                    {
+                      if (std::find(phase_column_names.begin(),
+                                    phase_column_names.end(),
+                                    column_name) != phase_column_names.end())
+                        {
+                          AssertThrow(false,
+                                      ExcMessage("The PerpleX lookup file " + filename + " must have unique column names. "
+                                                 "Sometimes, the same phase is stable with >1 composition at the same "
+                                                 "pressure and temperature, so you may see several columns with the same name. "
+                                                 "Either combine columns with the same name, or change the names."));
+                        }
+                      else
+                        {
+                          // Populate phase_column_names with the column name
+                          // and phase_column_indices with the column index in the current lookup file
+                          // This makes populating the phase_volume_fractions object easier.
```

Comment:
Can you explain this a bit more? (Why? Easier than what?)

Reply:
I've removed this line; it used to make sense when I had the map, but not any more!